File: model/monetization/ProductModel.java
package com.digitinary.dgate.model.monetization;

import com.digitinary.dgate.model.RestrictionType;
import com.digitinary.dgate.model.apispec.ApiSpec;
import com.digitinary.dgate.model.runtimeconfig.RuntimeConfigPublicationModel;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import java.time.LocalDateTime;
import java.util.List;
import java.util.Set;

/**
 * Monetization product model.
 *
 * @author Salah Abu Msameh
 * @since 08/05/2024
 */
@JsonIgnoreProperties(ignoreUnknown = true)
@JsonInclude(JsonInclude.Include.NON_NULL)
@NoArgsConstructor
@AllArgsConstructor
@Setter
@Getter
@Builder
public class ProductModel {

    private Long productId;
    private String name;
    private String desc;
    private ProductStatus status;
    private String segment;
    private String version;
    private LocalDateTime createdAt;
    private LocalDateTime lastUpdatedAt;
    private boolean premium;
    private RestrictionType restriction;
    private boolean availableOnDevPortal;
    private boolean publishedOnProd;
    private List<ApiSpec> apiSpecs;
    private Set<ProductPlanPriceModel> productPlanPrices;
    private List<AssociatedResource> associatedResources;
    private List<RuntimeConfigPublicationModel> publications;

}

File: model/monetization/ProductModel.py
class ProductModel:
    def __init__(self):
        self.productId = None  # Long
        self.name = None  # String
        self.desc = None  # String
        self.status = None  # ProductStatus  # This might be an enum
        self.segment = None  # String
        self.version = None  # String
        self.createdAt = None  # LocalDateTime
        self.lastUpdatedAt = None  # LocalDateTime
        self.premium = None  # boolean
        self.restriction = None  # RestrictionType  # This might be an enum
        self.availableOnDevPortal = None  # boolean
        self.publishedOnProd = None  # boolean
        self.apiSpecs = None  # List<ApiSpec>
        self.productPlanPrices = None  # Set<ProductPlanPriceModel>
        self.associatedResources = None  # List<AssociatedResource>
        self.publications = None  # List<RuntimeConfigPublicationModel>